lib/ansible/modules/cloud/cloudstack/cs_physical_network.py

```diff
@@ -0,0 +1,478 @@
+
+        if not self.module.check_mode:
+            resource = self.query_api('createPhysicalNetwork', **args)
+
+            poll_async = self.module.params.get('poll_async')
+            if poll_async:
+                self.network = self.poll_job(resource, 'physicalnetwork')
+
+        return self.network
+
+    def _update_network(self):
+        network = self.get_physical_network()
+
+        args = dict(id=network['id'])
+        args.update(self._get_common_args())
+
+        if self.has_changed(args, network):
+            self.result['changed'] = True
+
+            if not self.module.check_mode:
+                resource = self.query_api('updatePhysicalNetwork', **args)
+
+                poll_async = self.module.params.get('poll_async')
+                if poll_async:
+                    self.physical_network = self.poll_job(resource, 'physicalnetwork')
+        return self.physical_network
+
+    def absent_network(self):
+        physical_network = self.get_physical_network()
+        if physical_network:
+            self.result['changed'] = True
+            args = {
+                'id': physical_network['id'],
+            }
+            if not self.module.check_mode:
+                resource = self.query_api('deletePhysicalNetwork', **args)
+                poll_async = self.module.params.get('poll_async')
+                if poll_async:
+                    self.poll_job(resource, 'success')
+
+        return physical_network
+
+
+def main():
+    argument_spec = cs_argument_spec()
+    argument_spec.update(dict(
+        physical_network=dict(required=True, aliases=['name']),
+        zone=dict(),
+        domain=dict(),
+        vlan=dict(),
+        nsps_disabled=dict(type='list'),
+        nsps_enabled=dict(type='list'),
+        network_speed=dict(choices=['1G', '10G']),
+        broadcast_domain_range=dict(choices=['POD', 'ZONE']),
+        isolation_method=dict(choices=['VLAN', 'GRE', 'L3']),
+        state=dict(choices=['present', 'enabled', 'disabled', 'absent'], default='present'),
+        tags=dict(aliases=['tag']),
+        poll_async=dict(type='bool', default=True),
+    ))
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        required_together=cs_required_together(),
+        supports_check_mode=True
+    )
+
+    acs_network = AnsibleCloudStackPhysicalNetwork(module)
+    state = module.params.get('state')
+    nsps_disabled = module.params.get('nsps_disabled', [])
+    nsps_enabled = module.params.get('nsps_enabled', [])
+
+    if state in ['absent']:
+        network = acs_network.absent_network()
+    else:
+        network = acs_network.present_network()
+    if nsps_disabled is not None:
+        for name in nsps_disabled:
+            acs_network.update_nsp(name=name, state='Disabled')
+
+    if nsps_enabled is not None:
+        for nsp_name in nsps_enabled:
+            if nsp_name in ['VirtualRouter', 'VpcVirtualRouter']:
```

Comment:
do you know if the name are case sensitive? AFAIK their are not, so we could change it in a way, any thoughts?

~~~python
if nsp_name.lower() in ['virtualrouter', 'vpcvirtualrouter']:
~~~

Reply:
The names aren't used directly in API calls so it makes sense to make them case insensitive. Implemented in the latest commit.